src/main/java/uk/gov/hmcts/reform/judicialbooking/util/ValidationUtil.java

```diff
@@ -77,4 +79,100 @@ public static boolean validateTTL(String strDate) {
+        }
+        return true;
+    }
+
+    public static BookingEntity validateBookingRequest(BookingEntity booking) throws ParseException {
+        validateAppointmentId(booking.getAppointmentId());
+        validateBeginAndEndDates(booking);
+        return booking;
+    }
+
+    public static boolean validateAppointmentId(String appointmentId) {
+        if (appointmentId.isBlank()) {
+            throw new BadRequestException("The Booking appointmentId is invalid or missing.");
+        } else {
+            return true;
+        }
+    }
+
+    public static void validateBeginAndEndDates(BookingEntity booking) throws ParseException {
+        if (booking.getBeginTime() != null) {
+            if (booking.getBeginTime().getYear() == 1970) {
+                throw new BadRequestException(V1.Error.BAD_REQUEST_INVALID_DATETIME + " for beginTime");
+            }
+            validateDateTime(booking.getBeginTime().toString(), "BeginTime");
+        }
+        if (booking.getEndTime() != null) {
+            if (booking.getEndTime().getYear() == 1970) {
+                throw new BadRequestException(V1.Error.BAD_REQUEST_INVALID_DATETIME + " for endTime");
+            }
+            validateDateTime(booking.getEndTime().toString(), "EndTime");
+        }
+        if (booking.getBeginTime() != null && booking.getEndTime() != null) {
+            compareDateOrder(
+                    booking.getBeginTime().toString(),
+                    booking.getEndTime().toString()
+            );
+        }
+    }
+
+    public static void validateDateTime(String strDate, String timeParam) {
+        LOG.info("validateDateTime");
+        if (strDate.length() < 16) {
+            throw new BadRequestException(String.format(
+                    "Incorrect date format %s",
+                    strDate
+            ));
+        }
+        SimpleDateFormat simpleDateFormat = new SimpleDateFormat(Constants.DATE_PATTERN);
+        simpleDateFormat.setLenient(false);
+        Date javaDate;
+        try {
+            javaDate = simpleDateFormat.parse(strDate);
+            if (LOG.isInfoEnabled() && javaDate != null) {
+                LOG.info(javaDate.toString());
+            }
+        } catch (ParseException e) {
+            throw new BadRequestException(String.format(
+                    "Incorrect date format %s",
+                    strDate
+            ));
+        }
+        assert javaDate != null;
+        if (javaDate.before(new Date())) {
+            throw new BadRequestException(String.format(
+                    "The parameter '%s' cannot be prior to current date", timeParam
+            ));
+        }
+    }
+
+    public static void compareDateOrder(String beginTime, String endTime) throws ParseException {
+        SimpleDateFormat sdf = new SimpleDateFormat(Constants.DATE_PATTERN);
+        Date beginTimeP = sdf.parse(beginTime);
+        Date endTimeP = sdf.parse(endTime);
+        Date createTimeP = new Date();
+
```

Comment:
we have first converted dates into string and then again converted it into date with specific format. Can't we directly send the dates and compare it without parsing the date from strings?

Reply:
yeah I shouldn't have just copied this one over, thanks